src/Compilers/CSharp/Portable/BoundTree/BoundExpressionExtensions.cs

```diff
@@ -247,10 +247,16 @@ internal static TypeSymbolWithAnnotations GetTypeAndNullability(this BoundExpres
                         var conversion = (BoundConversion)expr;
                         if (conversion.ExplicitCastInCode)
                         {
-                            return conversion.IsExplicitlyNullable;
+                            if (conversion.IsExplicitlyNullable)
+                            {
+                                return true;
+                            }
+                            else if (expr.ConstantValue == null)
+                            {
+                                return false;
```

Comment:
Consider adding a comment

Reply:
Reverted change.

---
In reply to: [186881628](https://github.com/dotnet/roslyn/pull/26726#discussion_r186881628) [](ancestors = 186881628)

Reply:
> Reverted change.

Could you please come up with a test that would fail if you kept the change to ```internal static bool? IsNullable(this BoundExpression expr)```

---
In reply to: [187244748](https://github.com/dotnet/roslyn/pull/26726#discussion_r187244748) [](ancestors = 187244748,186881628)

Reply:
I've not found a code path yet that depends on the difference in implementation so I don't have a test at this time. I'll keep it in mind though.

---
In reply to: [187426126](https://github.com/dotnet/roslyn/pull/26726#discussion_r187426126) [](ancestors = 187426126,187244748,186881628)